src/Abstractions/NexusMods.Abstractions.Loadouts.Synchronizers/Rules/ActionMapping.cs

```diff
@@ -0,0 +1,122 @@
+            AxB_xxX_I => BackupFile | DeleteFromDisk | ExtractToDisk,
+            AxB_XxX_I => BackupFile | DeleteFromDisk | ExtractToDisk,
+            AAx_xxx_i => BackupFile | DeleteFromDisk,
+            AAx_XXx_i => DeleteFromDisk,
+            AAx_xxx_I => BackupFile | DeleteFromDisk,
+            AAx_XXx_I => DeleteFromDisk,
+            AAA_xxx_i => BackupFile,
+            AAA_XXX_i => DoNothing,
+            AAA_xxx_I => DoNothing,
+            AAA_XXX_I => DoNothing,
+            AAB_xxx_i => WarnOfUnableToExtract,
+            AAB_XXx_i => WarnOfUnableToExtract,
+            AAB_xxX_i => BackupFile | DeleteFromDisk | ExtractToDisk,
+            AAB_XXX_i => DeleteFromDisk | ExtractToDisk,
+            AAB_xxx_I => WarnOfUnableToExtract,
+            AAB_XXx_I => WarnOfUnableToExtract,
+            AAB_xxX_I => BackupFile | DeleteFromDisk | ExtractToDisk,
+            AAB_XXX_I => DeleteFromDisk | ExtractToDisk,
+            ABx_xxx_i => BackupFile | DeleteFromDisk,
+            ABx_Xxx_i => DeleteFromDisk,
+            ABx_xXx_i => BackupFile | DeleteFromDisk,
+            ABx_XXx_i => DeleteFromDisk,
+            ABx_xxx_I => BackupFile | DeleteFromDisk,
+            ABx_Xxx_I => DeleteFromDisk,
+            ABx_xXx_I => BackupFile | DeleteFromDisk,
+            ABx_XXx_I => DeleteFromDisk,
+            ABA_xxx_i => BackupFile,
+            ABA_XxX_i => DoNothing,
+            ABA_xXx_i => BackupFile,
+            ABA_XXX_i => DoNothing,
+            ABA_xxx_I => DoNothing,
+            ABA_XxX_I => DoNothing,
+            ABA_xXx_I => DoNothing,
+            ABA_XXX_I => DoNothing,
+            ABB_xxx_i => BackupFile | IngestFromDisk,
+            ABB_Xxx_i => IngestFromDisk,
+            ABB_xXX_i => BackupFile | IngestFromDisk,
+            ABB_XXX_i => IngestFromDisk,
+            ABB_xxx_I => BackupFile | IngestFromDisk,
+            ABB_Xxx_I => IngestFromDisk,
+            ABB_xXX_I => BackupFile | IngestFromDisk,
+            ABB_XXX_I => IngestFromDisk,
+            ABC_xxx_i => WarnOfConflict,
+            ABC_Xxx_i => WarnOfUnableToExtract,
+            ABC_xXx_i => WarnOfConflict,
+            ABC_xxX_i => BackupFile | IngestFromDisk,
+            ABC_XXx_i => BackupFile | IngestFromDisk,
+            ABC_XxX_i => WarnOfConflict,
+            ABC_xXX_i => BackupFile | IngestFromDisk,
+            ABC_XXX_i => IngestFromDisk,
+            ABC_xxx_I => BackupFile | IngestFromDisk,
+            ABC_Xxx_I => WarnOfUnableToExtract,
+            ABC_xXx_I => WarnOfUnableToExtract,
+            ABC_xxX_I => BackupFile | DeleteFromDisk | ExtractToDisk,
+            ABC_XXx_I => WarnOfUnableToExtract,
+            ABC_XxX_I => DeleteFromDisk | ExtractToDisk,
+            ABC_xXX_I => BackupFile | DeleteFromDisk | ExtractToDisk,
+            ABC_XXX_I => DeleteFromDisk | ExtractToDisk,
+            _ => throw new ArgumentOutOfRangeException(nameof(shorthand), shorthand, null),
```

Comment:
True, but if we ever add a new enum member this will blow up, where the `Enum.IsDefined` method will silently fail. I suppose we'll still get a `no switch case for X` error or whatever it's called. Is there a reason not to have the default leg for the switch?

Reply:
If you have a default case, and we add/remove an enum member, you get a suggestion in your IDE.

If you don't have a default case, and we add/remove an enum member, you get an error during compilation.

The compiler is smart enough to know that we cover all cases, so it doesn't complain or even suggest adding a new case.